File: 프로그래머스/0/181929. 원소들의 곱과 합/원소들의 곱과 합.java
class Solution {
    public int solution(int[] num_list) {
        int answer = 0;
        int product = 1;
        int sum_squared = 0;
        for (int num : num_list) {
            sum_squared += num;
            product *= num;
        }
        sum_squared = sum_squared * sum_squared;
        if (product < sum_squared) {
            answer = 1;
        }
        return answer;
    }
}

File: 프로그래머스/0/181929. 원소들의 곱과 합/원소들의 곱과 합.js
function solution(num_list) {
    var answer = 0;
    let product = num_list.reduce((acc, i) => acc * i, 1);
    let sum = num_list.reduce((acc, i) => acc + i, 0);
    let sum_squared = sum * sum;
    if (product < sum_squared) {
        answer = 1;
    }
    return answer;
}